### PR TITLE
Migrate to GitHub Actions

### LINKED ISSUE #435: Migrate to GitHub Actions
Most WPILib repos have already migrated, this one hasn't. A `.travis.yml` file exists even though we haven't used TravisCI for quite a while. I can work on a migration PR soon if wanted.

### PR DESCRIPTION
**Pull request by Starlight220**
Resolves #435 

@AustinShalit I'm not sure I used correct equivalents of the Azure Tasks (`Gradle`, `CopyFiles`, and `PublishBuildArtifacts`). Any chance you can take a look?

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    runs-on: ubuntu-latest
+
+    # This grabs the WPILib docker container
+    container: wpilib/roborio-cross-ubuntu:2020-18.04
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+    # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+    - uses: actions/checkout@v2
+
```
Comment (AustinShalit) [inline .github/workflows/ci.yml:27]:
```suggestion
    - uses: actions/checkout@v2
       with:
         fetch-depth: 0
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    - name: Publish to Maven Local
+      run: |
+        chmod +x gradlew
+        ./gradlew -Xmx3072m build publishToMavenLocal PatchExamples --stacktrace -PlocalPublish
```
Comment (AustinShalit) [inline .github/workflows/ci.yml:31]:
```suggestion
      run: ./gradlew build publishToMavenLocal PatchExamples --stacktrace -PlocalPublish
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      
+      
+    - name: Test Cpp Build
+      run: |
+        cd examples/cpp
+        chmod +x ./gradlew
+        ./gradlew build
```
Comment (AustinShalit) [inline .github/workflows/ci.yml:38]:
```suggestion
      run: ./gradlew build
      working-directory: examples/cpp
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        
+    - name: Test Java Build
+      run: |
+        cd examples/java
+        chmod +x ./gradlew
+        ./gradlew build
```
Comment (AustinShalit) [inline .github/workflows/ci.yml:44]:
```suggestion
      run: ./gradlew build
      working-directory: examples/java
```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        
+        
+    - name: Test JNI Build
+      run: |
+        cd examples/jni
+        chmod +x ./gradlew
+        ./gradlew build
```
Comment (AustinShalit) [inline .github/workflows/ci.yml:51]:
```suggestion
      run: ./gradlew build
      working-directory: examples/jni
```

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+        
+    - name: Test Kotlin Build
+      run: |
+        cd examples/kotlin
+        chmod +x ./gradlew
+        ./gradlew build
```
Comment (AustinShalit) [inline .github/workflows/ci.yml:57]:
```suggestion
      run: ./gradlew build
      working-directory: examples/kotlin
```

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+        
+    - uses: actions/upload-artifact@v2
+      with:
+          name: "Examples"
```
Comment (AustinShalit) [inline .github/workflows/ci.yml:61]:
```suggestion
    - uses: actions/upload-artifact@v2
      with:
        name: Examples
        path: build/distributions
```

**Review by AustinShalit — CHANGES_REQUESTED** [review]

**Comment by AustinShalit** [comment]
Thank you @Starlight220 I've left some comments just to make the file more succinct.

File: .github/workflows/ci.yml [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+      
+    - name: Test Cpp Build
+      run: ./gradlew build
+      working-directory: examples/cpp
+        
+    - name: Test Java Build
+      run: ./gradlew build
+      working-directory: examples/java
+        
+
```
Comment (AustinShalit) [inline .github/workflows/ci.yml:25]:
```suggestion
```

File: .github/workflows/ci.yml [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+    runs-on: ubuntu-latest
+    container: wpilib/roborio-cross-ubuntu:2020-18.04
+    steps:
+    - uses: actions/checkout@v2
+       with:
+         fetch-depth: 0
+    - name: Publish to Maven Local
+      run: ./gradlew build publishToMavenLocal PatchExamples --stacktrace -PlocalPublish
+      
+
```
Comment (AustinShalit) [inline .github/workflows/ci.yml:16]:
```suggestion
```

File: .github/workflows/ci.yml [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    container: wpilib/roborio-cross-ubuntu:2020-18.04
+    steps:
+    - uses: actions/checkout@v2
+       with:
+         fetch-depth: 0
```
Comment (AustinShalit) [inline .github/workflows/ci.yml:12]: I think the spacing is wrong here but can't fix it in the GH editor.

**Comment by AustinShalit** [comment]
Please also update the README badge and delete the azure and travis files.

**Comment by Starlight220** [comment]
Nope. Why isn't it running?

**Comment by Starlight220** [comment]
Is there a reason it would fail because of missing permissions only on the JNI build and not on the Cpp and Java ones? 
Kotlin etc doesn't get tested because of the JNI failure.

File: .github/workflows/ci.yml [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+        ls -al ~/.m2
+    - uses: actions/upload-artifact@v2.2.0
+      with:
+        name: Maven
+        path: ~/.m2/
+      
+  test_examples:
+    needs: build
+    runs-on: ubuntu-latest
+#     container: wpilib/roborio-cross-ubuntu:2020-18.04
```
Comment (AustinShalit) [inline .github/workflows/ci.yml:29]:
```suggestion
```

File: .github/workflows/ci.yml [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+      with:
+       fetch-depth: 0
+    - uses: actions/setup-java@v1
+      with:
+        java-version: 11
+    - name: Publish to Maven Local
+      run: ./gradlew build publishToMavenLocal --stacktrace -PlocalPublish
+    - name: Test (delete this)
+      run: |
+        ls -al ~/.m2
```
Comment (AustinShalit) [inline .github/workflows/ci.yml:20]:
```suggestion
```

File: .github/workflows/ci.yml [hunk 13]
```diff
@@ -0,0 +1,54 @@
+name: CI
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+#     container: wpilib/roborio-cross-ubuntu:2020-18.04
```
Comment (AustinShalit) [inline .github/workflows/ci.yml:8]:
```suggestion
```

**Review by AustinShalit — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+    runs-on: ubuntu-latest
+    strategy:
+     matrix:
+        language: [cpp, java, jni, kotlin]
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-java@v1
+      with:
+        java-version: 11
+    - uses: actions/download-artifact@v2.0.5
```
Comment (AustinShalit) [inline .github/workflows/ci.yml:33]:
```suggestion
    - uses: actions/download-artifact@v2
```

File: .github/workflows/ci.yml [hunk 15]
```diff
@@ -0,0 +1,49 @@
...
+    steps:
+    - uses: actions/checkout@v2
+      with:
+       fetch-depth: 0
+    - uses: actions/setup-java@v1
+      with:
+        java-version: 11
+    - name: Publish to Maven Local
+      run: ./gradlew build publishToMavenLocal --stacktrace -PlocalPublish
+    - uses: actions/upload-artifact@v2.2.0
```
Comment (AustinShalit) [inline .github/workflows/ci.yml:17]:
```suggestion
    - uses: actions/upload-artifact@v2
```

File: README.md [hunk 16]
```diff
@@ -1,3 +1,4 @@
+![CI](https://github.com/wpilibsuite/GradleRIO/workflows/CI/badge.svg)
```
Comment (AustinShalit) [inline README.md:1]:
```suggestion
![CI](https://github.com/wpilibsuite/GradleRIO/workflows/CI/badge.svg)

```

**Review by AustinShalit — CHANGES_REQUESTED** [review]
Sorry @Starlight220 just a few more things.

**Review by AustinShalit — APPROVED** [review]

**Pull request merged**